src/main/java/dk/kb/storage/storage/DsStorage.java

```diff
@@ -365,6 +381,46 @@ public ArrayList<DsRecordDto > getModifiedAfterParentsOnly(String origin, long m
         return records; 
     }
 
+    
+
+    /**
+     * <p>
+     * Get a list of records after a given mTime. The records will only have fields
+     * id,mTime,referenceid and kalturaid defined 
+     * </p>
+     *
+     *@param origin The origin to fetch records drom
+     *@param mTime only fetch records with mTime larger that this
+     *@param batchSize Number of maximum records to return
+     *
+     * @return List of records only have fields id,mTime,referenceid and kalturaid
+     */
+    public ArrayList<DsRecordReferenceIdDto> getReferenceIds(String origin, long mTime, int batchSize) throws Exception {
+
+        if (batchSize <1 || batchSize > 100000) { //No doom switch
+            throw new Exception("Batchsize must be in range 1 to 100000");          
+        }
+        ArrayList<DsRecordReferenceIdDto> records = new ArrayList<DsRecordReferenceIdDto>();
+        try (PreparedStatement stmt = connection.prepareStatement(referenceIdsStatement)) {
+
+            stmt.setString(1, origin);
+            stmt.setLong(2, mTime);
+            stmt.setLong(3, batchSize);
+                        
+            try (ResultSet rs = stmt.executeQuery();) {
+                while (rs.next()) {
+                    DsRecordReferenceIdDto  record = createRecordReferenceIdFromRS(rs);
+                    records.add(record);
+                }
+            }
+        }
+        catch(Exception e) {
+            throw new Exception("SQL error getReferenceIds",e);
+        }
```

Comment:
Maybe catch and throw the more narrow SQLException

Reply:
changed